src/snowflake/snowpark/session.py

```diff
@@ -265,6 +265,9 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
+    def __repr__(self):
+        return f"<{self.__class__.__module__}.{self.__class__.__name__}: account={repr(self._conn._conn.account)}, role={repr(self._conn._conn.role)}, database={repr(self._conn._conn.database)}, schema={repr(self._conn._conn.schema)}, warehouse={repr(self._conn._conn.warehouse)}>"
```

Comment:
1. `self._conn._conn.role` and other similar properties can be None. You should use `get_current_*` methods in session class
2. I don't think you need `repr(x)` then.
3. Add tests when these properties are None (`get_current_*` methods  can also return None).

Reply:
When can roles and other properties be empty? It seems to me that the `snowflake-connector-python` maintains the contents of these variables while logging in and executing SQL queries.
https://github.com/snowflakedb/snowflake-connector-python/blob/d0efd0f8333db210a3c482b6f631fe4ba01eabdc/src/snowflake/connector/connection.py#L1013-L1026

Reply:
1+2. I will use change the implementation to use `get_current_*` for better consistency. 
3. I agree with @sfc-gh-kbregula that for our use case, `get_current_*` will always short circuit at `getattr(self._conn, param) ` and return default values that the Python connector connection store.

Reply:
When we create a session in Snowflake, database/schema/warehouse is optional. If we don't set them in connection parameters of python connector, then these properties will be empty.

Reply:
nope, it's possible that the user doesn't have a default database/schema/warehouse. (For example, if you log into snowflake at the first time and don't create them before, how can we have defaults?)

Reply:
I see, that makes sense 👍